rotkehlchen/exchanges/binance.py

```diff
@@ -589,6 +607,184 @@ def _query_lending_balances(
+        range_query_name = f'{self.location}_history_events_{self.name}'
+        ranges_to_query = ranges.get_location_query_ranges(
+            cursor=cursor,
+            location_string=range_query_name,
+            start_ts=start_ts,
+            end_ts=end_ts,
+        )
+        for query_start_ts, query_end_ts in ranges_to_query:
+            events = []
+            for lending_type in ('DAILY', 'ACTIVITY', 'CUSTOMIZED_FIXED'):
+                response = self._api_query_list_paginated_within_time_delta(
+                    api_type='sapi',
+                    start_ts=query_start_ts,
+                    time_delta=BINANCE_LENDING_INTEREST_HISTORY_INTERVAL_TS,
+                    end_ts=query_end_ts,
+                    method='lending/union/interestHistory',
+                    additional_options={'lendingType': lending_type, 'size': 100},
+                )
+                for idx, entry in enumerate(response):
+                    try:
+                        interest_received = deserialize_asset_amount_force_positive(entry['interest'])  # noqa: E501
+                        if interest_received == ZERO:
+                            continue
+
+                        event_identifier = HistoryBaseEntry.deserialize_event_identifier(str(uuid4()))  # noqa: E501
+                        timestamp = TimestampMS(
+                            deserialize_fval(
+                                value=entry['time'],
+                                name='time',
+                                location='binance lending interest processing',
+                            ).to_int(exact=False),
+                        )
+                        asset = asset_from_binance(entry['asset'])
+                    except (UnsupportedAsset, UnknownAsset) as e:
+                        error_type = 'unknown' if isinstance(e, UnknownAsset) else 'unsupported'
+                        self.msg_aggregator.add_warning(
+                            f'Found {error_type} {self.name} asset {e.identifier}. '
+                            f'Ignoring its lending interest history query.',
+                        )
+                        continue
+                    except KeyError as e:
+                        self.msg_aggregator.add_error(
+                            f'Missing key entry for {str(e)} in {self.name} {entry}. '
+                            f'Ignoring its lending interest history query.',
+                        )
+                        continue
+                    except DeserializationError as e:
+                        self.msg_aggregator.add_error(
+                            f'Error at deserializing {self.name} asset. {str(e)}. '
+                            f'Ignoring its lending interest history query.',
+                        )
+                        continue
+
+                    event = HistoryBaseEntry(
+                        event_identifier=event_identifier,
+                        sequence_index=idx,
```

Comment:
Since the event identifier is different for each one you can use 0 as sequence index for all. Sequence index is used when multiple historyBaseEntry share the same event identifier